src/Illuminate/Bus/PendingBatch.php

```diff
@@ -92,14 +99,16 @@ protected function ensureJobIsBatchable(object|array $job): void
     {
         foreach (Arr::wrap($job) as $job) {
             if ($job instanceof PendingBatch) {
-                $this->ensureJobIsBatchable($job->jobs->all());
-
                 return;
             }
 
-            if (! in_array(Batchable::class, class_uses_recursive($job))) {
+            if (! (static::$batchableClasses[$job::class] ?? false) && ! in_array(Batchable::class, class_uses_recursive($job))) {
```

Comment:
While it's not possible with `class_uses_recursive`, this could be taken one step further, by implementing something similar to `class_uses_recursive`: Caching all classes in the recursive chain 🤔

Reply:
Something along the lines of this: 3111c97

Reply:
or 1990987